test/Migrator/string-representable.swift.expected

```diff
@@ -66,6 +66,56 @@ func bar(_ c: Container) {
   c.add(single: attr)
 }
 
+public class SubContainer: Container {
+  public override func adding(optionalAttributes subname: [String: Any]?) {
+// Local variable inserted by Swift 4.2 migrator.
+let subname = convertFromOptionalSimpleAttributeDictionary(subname)
```

Comment:
Do we care about getting a compilation error if they've already shadowed the parameter for other reasons (e.g. to make it var)? I'd guess it's not that common.

Reply:
ah, that's definitely likely to happen. One option in that case is we avoid inserting the variable at all; however, it will leave other compiler errors for using the original parameter's type...